Source/ThirdParty/scope_guard/scope_guard.hpp

```diff
@@ -0,0 +1,177 @@
+/*
+ *  Created on: 13/02/2018
+ *      Author: ricab
+ *
+ * See README.md for documentation of this header's public interface.
+ */
+
+#ifndef SCOPE_GUARD_HPP_
+#define SCOPE_GUARD_HPP_
+
+#include <type_traits>
+#include <utility>
+
+#if __cplusplus >= 201703L && defined(SG_REQUIRE_NOEXCEPT_IN_CPP17)
```

Comment:
In C++17, exception specifications are part of the type. That allows rejecting, at compile time, callbacks that throw - to avoid throwing from a destructor.

Reply:
https://github.com/ricab/scope_guard#compilation-option-sg_require_noexcept_in_cpp17